stl/inc/deque

```diff
@@ -1473,47 +1473,34 @@ private:
 
 public:
     void pop_front() noexcept /* strengthened */ {
+#if _MSVC_STL_HARDENING_DEQUE || _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(!empty(), "pop_front() called on empty deque");
+#endif
+
 #if _ITERATOR_DEBUG_LEVEL == 2

```

Comment:
Should this be included in STL hardening too?
`#if _MSVC_STL_HARDENING_DEQUE || _ITERATOR_DEBUG_LEVEL == 2`

Reply:
Good question, given the similar-looking code patterns. The answer is no: here, `IDL == 2` guards debug-only code for fine-grained iterator invalidation, which requires ABI-breaking extra data members to be added to containers and iterators. It cannot be enabled for hardening.